fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ReferencesRdfContext.java

```diff
@@ -48,21 +57,36 @@ public ReferencesRdfContext(final FedoraResource resource,
         throws RepositoryException {
         super(resource, idTranslator);
         property2triple = new PropertyToTriple(resource.getNode().getSession(), idTranslator);
-        concat(putStrongReferencePropertiesIntoContext());
-        concat(putWeakReferencePropertiesIntoContext());
+        session = resource.getNode().getSession();
+        putReferencesIntoContext(resource.getNode().getWeakReferences());
+        putReferencesIntoContext(resource.getNode().getReferences());
     }
 
-    private Iterator<Triple> putWeakReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getWeakReferences();
+    private void putReferencesIntoContext(final Iterator<Property> properties) throws RepositoryException {
+        while (properties.hasNext()) {
```

Comment:
Ok, no problem. Look at what is done in `PropertiesRdfContext`. The function mapped over an iteration of properties there goes from a property to an iterator of triples. (This is just flatmapping.) So if there are multiple triples from a property, no problem. If there is only one, no problem (iterator of one). And if there are none, no problem (empty iterator). This is a handy trick for this kind of situation-- rely on the fact that the "optional" monad is an example of a "collectionish" monad.